SPECS/glibc/glibc.spec

```
@@ -209,6 +209,14 @@ popd
 sed -i 's@#! /bin/bash@#! /bin/sh@' %{buildroot}%{_bindir}/ldd
 sed -i 's@#!/bin/bash@#!/bin/sh@' %{buildroot}%{_bindir}/tzselect
 
+# Determine which static libs are needed in `glibc-devel` - the rest will be put
+# into `glibc-static`.  We need to keep the static shims for function that's now
+# in `libc.so` (since 2.34 - see https://developers.redhat.com/articles/2021/12/17/why-glibc-234-removed-libpthread)
+# and the "statically linked bit" of `libc.so` (called `libc_nonshared.a`)
+static_libs_in_devel_pattern="lib\(c_nonshared\|pthread\|dl\|rt\|g\|util\|mcheck\).a"
+ls -1 %{buildroot}%{_lib64dir}/*.a | grep -e "$static_libs_in_devel_pattern" | sed "s:^%{buildroot}::g" > devel.filelist
+ls -1 %{buildroot}%{_lib64dir}/*.a | grep -v -e "$static_libs_in_devel_pattern" | sed "s:^%{buildroot}::g" > static.filelist
```

Idea to remove most of the `\` to make the pattern a bit easier to read:

```suggestion
static_libs_in_devel_pattern="lib(c_nonshared|pthread|dl|rt|g|util|mcheck)\.a"
ls -1 %{buildroot}%{_lib64dir}/*.a | grep -P "$static_libs_in_devel_pattern" | sed "s:^%{buildroot}::g" > devel.filelist
ls -1 %{buildroot}%{_lib64dir}/*.a | grep -v -P "$static_libs_in_devel_pattern" | sed "s:^%{buildroot}::g" > static.filelist
```

Another one to drop `sed`:

```suggestion
static_libs_in_devel_pattern="(?<=%{buildroot}).*lib(c_nonshared|pthread|dl|rt|g|util|mcheck)\.a"
ls -1 %{buildroot}%{_lib64dir}/*.a | grep -oP "$static_libs_in_devel_pattern" > devel.filelist
ls -1 %{buildroot}%{_lib64dir}/*.a | grep -v -oP "$static_libs_in_devel_pattern" > static.filelist
```

Feel free to completely ignore both, though. ;)